app/src/main/java/protect/card_locker/importexport/CatimaImporter.java

```diff
@@ -381,28 +496,61 @@ private void importLoyaltyCard(SQLiteDatabase database, CSVRecord record)
      * Import a single group into the database using the given
      * session.
      */
-    private void importGroup(SQLiteDatabase database, CSVRecord record) throws FormatException {
-        String id = CSVHelpers.extractString(DBHelper.LoyaltyCardDbGroups.ID, record, null);
+    private void importGroupV2(SQLiteDatabase database, CSVRecord record) throws FormatException {
+        String name = CSVHelpers.extractString(DBHelper.LoyaltyCardDbGroups.ID, record, null);
 
-        if (id == null) {
-            throw new FormatException("Group has no ID: " + record);
+        if (name == null) {
+            throw new FormatException("Group has no name: " + record);
         }
 
-        DBHelper.insertGroup(database, id);
+        DBHelper.insertGroup(database, name);
+    }
+
+    /**
+     * Import groups from the V3 scheme (database v17) into the database using the given session.
+     */
+    private void importGroupsV3(SQLiteDatabase database, List<CSVRecord> records) throws FormatException {
+        TreeMap<Integer, Group> sortedGroups = new TreeMap<Integer, Group>();
+        for (CSVRecord record : records) {
+            int id = CSVHelpers.extractInt(DBHelper.LoyaltyCardDbGroups.ID, record);
+            String name = CSVHelpers.extractString(DBHelper.LoyaltyCardDbGroups.NAME, record, null);
+            int order = CSVHelpers.extractInt(DBHelper.LoyaltyCardDbGroups.ORDER, record);
+
+            if (name == null) {
+                throw new FormatException("Group has no name: " + record);
+            }
+
+            DBHelper.insertGroup(database, id, name);
```

Comment:
We should check for duplicates here, and re-use an existing group with the same name if it already exists.